notifications/notifications.py

```diff
@@ -0,0 +1,118 @@
+import os
```

Comment:
Small nitpick! 

Perhaps a docstring at the top of the page for some extra documentation?

Looks really good, and well optimised otherwise! Great work